server/controllers/threads.controller.js

```diff
@@ -185,40 +225,67 @@ function show(req, res, next) {
     .catch(e => next(e));
 }
 
+
 /**
  * Returns a the current user and a list of the current user's threads
  * @returns {User}
  */
 function index(req, res, next) {
   const { username } = req.user;
-  User.findOne({
-    where: {username}
+  const { logUsername } = req.query;
+
+
+  if( username === logUsername ) {
+    sequelize.query('SELECT * FROM "Users" as A inner join "UserThreads" as UserThread on A."id" = UserThread."UserId" inner join "Threads" as E on UserThread."ThreadId" = E."id" and (E."logUserId" = A."id" OR E."logUserId" is null ) inner join "Messages" as LastMessage on E."lastMessageId" = LastMessage."id" Where A.username = :username Order By "lastMessageSent" DESC',
```

Comment:
@nickmccally do you think native Sequelize methods can handle these queries? If so it may be better to resort to sequelize query methods as that would make the code adaptable to any SQL engine.